src/pages/ReportSettingsPage.js

```diff
@@ -71,8 +73,8 @@ class ReportSettingsPage extends Component {
     updatePolicyRoomName(values) {
         Keyboard.dismiss();
 
-        // When the room name has not changed, skip the Form submission
-        if (values.newRoomName === this.props.report.reportName) {
+        // When the room name has not changed or should not be changed, skip the Form submission
+        if (this.shouldDisableRename || values.newRoomName === this.props.report.reportName) {
```

Comment:
Do we need this check here if we're already not showing the submit button?

Reply:
Even when the submit button is not being displayed, hitting `Enter` will still trigger `onSubmit`.